orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/v2schema/model/V2PipelineTemplate.java

```diff
@@ -0,0 +1,307 @@
+/*
+ * Copyright 2018 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.pipelinetemplate.v2schema.model;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.netflix.spinnaker.orca.pipelinetemplate.v1schema.model.NamedContent;
+import com.netflix.spinnaker.orca.pipelinetemplate.v1schema.model.NamedHashMap;
+import com.netflix.spinnaker.orca.pipelinetemplate.v1schema.model.StageDefinition;
+import com.netflix.spinnaker.orca.pipelinetemplate.validator.VersionedSchema;
+
+import java.util.*;
+
+public class V2PipelineTemplate implements VersionedSchema {
+
+  public static final String SCHEMA = "schema";
+  public static final String V2_SCHEMA_VERSION = "2";
+
+  private String schema;
+  private String id;
+  private Metadata metadata = new Metadata();
+  private Boolean protect = false;
```

Comment:
What does this do?

Reply:
Specifies whether a template's stage graph is mutable by a config, added a comment to clarify.